docs/en/contribution-roadmap/Knowledge.md

```diff
@@ -11,5 +11,36 @@ priority: 10
 ## Java
 
 ## Maven
+The Alluxio project uses [Maven](https://maven.apache.org/) to manage its dependencies and to build its source code. 
+Here we provide some simple commands to get you started.
+For a full tutorial on maven, please consult the maven documentation.
+
+### Build with Maven
+The default command to build with maven is through the `maven install` command.
+The following is a command that Alluxio developers often use to speed up the compilation. 
+
+```bash
+mvn -T 4C clean  install -Dmaven.javadoc.skip=true -DskipTests -Dlicense.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true  -Phadoop-2 -Dhadoop.version=2.7.3 -Pufs-hadoop-3  -Dufs.hadoop.version=3.3.0 -pl \!webui
+```
+
+The above command means to build using 4 cores and skip javadoc generation, skip tests, skip license header checks, skip checkstyle and findbugs checks, have alluxio export a hadoop-2 compatible interface with a hadoop version of 2.7.3, build it to support hadoop 3 as ufs with a version of 3.3.0, and finally skip the webui submodule (because it can take a while).
+
+### Dependency Analysis with Maven
+Aside from building the project, Maven is also responsible for managing the dependency tree for Alluxio.
+If dependencies are not managed properly, for example, if there are conflicting versions of the same package in the final jar file, it is possible to have run time failures or ClassNotFoundExceptions thrown. 
+
+To display the dependency tree for further analysis, use the following command. 
+
+```bash
+mvn dependency:tree
+```
+
```

Comment:
can you add a section about the maven build checks that are run by github actions? i found that most people will be able to get to the console log to see the BUILD FAILED at the bottom but it is not intuitive to find the specific failures.